cuda_core/cuda/core/experimental/_program.py

```diff
@@ -422,7 +423,326 @@ def __post_init__(self):
+            unsupported.append("device_code_optimize")
+        if self.use_fast_math is not None and self.use_fast_math:
+            unsupported.append("use_fast_math")
+        if self.extra_device_vectorization is not None and self.extra_device_vectorization:
+            unsupported.append("extra_device_vectorization")
+        if self.gen_opt_lto is not None and self.gen_opt_lto:
+            unsupported.append("gen_opt_lto")
+        if self.define_macro is not None:
+            unsupported.append("define_macro")
+        if self.undefine_macro is not None:
+            unsupported.append("undefine_macro")
+        if self.include_path is not None:
+            unsupported.append("include_path")
+        if self.pre_include is not None:
+            unsupported.append("pre_include")
+        if self.no_source_include is not None and self.no_source_include:
+            unsupported.append("no_source_include")
+        if self.std is not None:
+            unsupported.append("std")
+        if self.builtin_move_forward is not None:
+            unsupported.append("builtin_move_forward")
+        if self.builtin_initializer_list is not None:
+            unsupported.append("builtin_initializer_list")
+        if self.disable_warnings is not None and self.disable_warnings:
+            unsupported.append("disable_warnings")
+        if self.restrict is not None and self.restrict:
+            unsupported.append("restrict")
+        if self.device_as_default_execution_space is not None and self.device_as_default_execution_space:
+            unsupported.append("device_as_default_execution_space")
+        if self.device_int128 is not None and self.device_int128:
+            unsupported.append("device_int128")
+        if self.optimization_info is not None:
+            unsupported.append("optimization_info")
+        if self.no_display_error_number is not None and self.no_display_error_number:
+            unsupported.append("no_display_error_number")
+        if self.diag_error is not None:
+            unsupported.append("diag_error")
+        if self.diag_suppress is not None:
+            unsupported.append("diag_suppress")
+        if self.diag_warn is not None:
+            unsupported.append("diag_warn")
+        if self.brief_diagnostics is not None:
+            unsupported.append("brief_diagnostics")
+        if self.fdevice_syntax_only is not None and self.fdevice_syntax_only:
+            unsupported.append("fdevice_syntax_only")
+        if self.minimal is not None and self.minimal:
+            unsupported.append("minimal")
+        if self.numba_debug is not None and self.numba_debug:
+            unsupported.append("numba_debug")
+        
+        if unsupported:
+            raise CUDAError(
+                f"The following options are not supported by nvJitLink backend: {', '.join(unsupported)}"
+            )
+        
+        return list(o.encode() for o in options)
+
+    def _prepare_nvvm_options(self) -> list[str]:
```

Comment:
Please implement it like this
```suggestion
    def _prepare_nvvm_options(self, as_bytes: bool = True) -> Union[list[bytes], list[str]]:
```
- when `as_bytes` is True, return `list[bytes]`. Use this in `as_bytes()`.
- otherwise, return `list[str]`. Use this internally when `_prepare_nvvm_options` is called.